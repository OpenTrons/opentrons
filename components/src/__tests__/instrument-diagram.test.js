import React from 'react'
import Renderer from 'react-test-renderer'

import { InstrumentDiagram, InstrumentGroup } from '..'

describe('InstrumentDiagram', () => {
  test('Single-channel renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentDiagram channels={1} displayCategory="OG" />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Multi-channel renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentDiagram channels={8} displayCategory="OG" />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Single-channel GEN2 renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentDiagram channels={1} displayCategory="GEN2" />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Multi-channel GEN2 renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentDiagram channels={8} displayCategory="GEN2" />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('InstrumentGroup', () => {
  test('Renders correctly', () => {
    const tree = Renderer.create(
      <InstrumentGroup
        left={{
          mount: 'left',
          description: 'p300 8-Channel',
          tipType: '150',
          pipetteSpecs: { channels: 8, displayCategory: 'OG' },
          className: 'foo',
        }}
        right={{
          mount: 'right',
          description: 'p10 Single',
          tipType: '10',
          pipetteSpecs: { channels: 1, displayCategory: 'OG' },
          isDisabled: true,
          className: 'blah',
        }}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
