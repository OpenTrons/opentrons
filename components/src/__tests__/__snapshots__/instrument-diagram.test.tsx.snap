// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstrumentDiagram Multi-channel GEN2 renders correctly 1`] = `
<div>
  <img
    className="flipped_image"
    src="pipetteGEN2Multi.png"
  />
</div>
`;

exports[`InstrumentDiagram Multi-channel renders correctly 1`] = `
<div>
  <img
    className=""
    src="pipetteMulti.png"
  />
</div>
`;

exports[`InstrumentDiagram Single-channel GEN2 renders correctly 1`] = `
<div>
  <img
    className="flipped_image"
    src="pipetteGEN2Single.png"
  />
</div>
`;

exports[`InstrumentDiagram Single-channel renders correctly 1`] = `
<div>
  <img
    className=""
    src="pipetteSingle.png"
  />
</div>
`;

exports[`InstrumentGroup Renders correctly 1`] = `
<section
  className="pipette_group"
>
  <div
    className="pipette left foo"
  >
    <div
      className="pipette_info"
    >
      <div>
        <h2
          className="title"
        >
          pipette
        </h2>
        <span
          className="value"
        >
          p300 8-Channel
        </span>
      </div>
    </div>
    <div
      className="pipette_icon"
    >
      <img
        className=""
        src="pipetteMulti.png"
      />
    </div>
  </div>
  <div
    className="pipette right disabled blah"
  >
    <div
      className="pipette_info"
    >
      <div>
        <h2
          className="title"
        >
          pipette
        </h2>
        <span
          className="value"
        >
          p10 Single
        </span>
      </div>
    </div>
    <div
      className="pipette_icon"
    >
      <img
        className="flipped_image"
        src="pipetteSingle.png"
      />
    </div>
  </div>
</section>
`;
